src/Identity/Extensions.Core/src/StoreOptions.cs

```diff
@@ -19,4 +21,10 @@ public class StoreOptions
     /// This will be enforced by requiring the store to implement <see cref="IProtectedUserStore{TUser}"/>.
     /// </summary>
     public bool ProtectPersonalData { get; set; }
+
+    /// <summary>
+    /// The schema version for the store, the default is 0.0 which leaves it up to the store
+    /// to determine what version should be used.
+    /// </summary>
+    public Version SchemaVersion { get; set; } = new Version(0, 0);
```

Comment:
I don't think dotted version numbers make sense here. You would be better off with a monotonically increasing number since _minor_ changes imply some kind of compatibility between releases.

Reply:
Would we never consider making small schema version changes in older versions?  i.e. lets say we do 2.0 in 8, and 3.0 in 9.  We could continue to make small tweaks in 8 via 2.x, while if we just did an integer that would be harder unless we versioned using jumps to leave space i.e. 100, 200, 300

Reply:
Hrm, that is a good point. I hadn't considered it from a framework servicing point of view, more from a live-site ops point of view which is generally roll-forward.

It is an interesting question. Schema version increments probably only matter when you are making breaking changes to the schema (as opposed to just adding a new field). Assuming that holds true then we could probably make minor non-breaking changes to older frameworks without incrementing the versioning specifier provided we took that change into account for subsequent versions to make sure folks weren't cut off from their upgrade path.

Reply:
Well the idea is we always increment the version if we make any schema changes, otherwise we'd effectively only be doing semantic major versions.  I think its worth using Version over an int/string since the concept of major/minor version might be useful if we ever want to do minor versions in servicing